src/main/java/org/opensrp/web/config/security/filter/CrossSiteScriptingPreventionFilter.java

```diff
@@ -1,97 +1,51 @@
 package org.opensrp.web.config.security.filter;
 
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.node.ArrayNode;
-import com.fasterxml.jackson.databind.node.JsonNodeFactory;
-import com.fasterxml.jackson.databind.node.JsonNodeType;
-import com.fasterxml.jackson.databind.node.ObjectNode;
-import org.owasp.encoder.Encode;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpMethod;
 import org.springframework.http.MediaType;
 
 import javax.servlet.FilterChain;
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletResponse;
-import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletRequest;
 import javax.servlet.ServletException;
 import javax.servlet.Filter;
 import javax.servlet.FilterConfig;
 import java.io.IOException;
-import java.util.Iterator;
-import java.util.Map;
 
 public class CrossSiteScriptingPreventionFilter implements Filter {
 
 	private static Logger logger = LoggerFactory.getLogger(CrossSiteScriptingPreventionFilter.class);
-	private static ObjectMapper mapper = new ObjectMapper();
 
 	@Override
 	public void init(FilterConfig filterConfig) throws ServletException {
      // do nothing
 	}
-
-	@Override
-	public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
-			throws IOException, ServletException {
+	
+	public void doFilter(ServletRequest request, ServletResponse response,
+			FilterChain chain) throws IOException, ServletException {
 		logger.debug("Inside CrossSiteScriptingPreventionFilter  ...............");
-		XssPreventionResponseWrapper xssPreventionResponseWrapper = new XssPreventionResponseWrapper(
-				(HttpServletResponse) response);
-		chain.doFilter(request, xssPreventionResponseWrapper);
-		String responseString = xssPreventionResponseWrapper.getCaptureAsString();
-		try {
-			if (response.getContentType() != null && response.getContentType().contains(MediaType.APPLICATION_JSON_VALUE)) {
-					JsonNode jsonNode = mapper.readTree(responseString);
-					JsonNode updatedJsonNode = encode(jsonNode);
-					String encodedJsonString = updatedJsonNode.toString();
-					response.getOutputStream().write(encodedJsonString.getBytes());
-			} 
-			else {
-				response.getOutputStream().write(responseString.getBytes());
-			}
-		}
-		catch (Exception e) {
-			logger.error("Exception occurred while sanitizing JSON response");
+		XssPreventionRequestWrapper wrappedRequest = new XssPreventionRequestWrapper(
+				(HttpServletRequest) request);
+
+		if ((wrappedRequest.getMethod().equals(HttpMethod.POST.name()) || wrappedRequest.getMethod()
+				.equals(HttpMethod.PUT.name()))
+				&& request.getContentType() != null && !request.getContentType()
+				.contains(MediaType.MULTIPART_FORM_DATA_VALUE)) {
+			wrappedRequest.resetInputStream();
```

Comment:
This method, is not required since `getInputStream` and `getReader` have been overriden

Reply:
Fixed.